src/main/kotlin/cacophony/automata/minimalization/DFAUtils.kt

```diff
@@ -72,7 +77,10 @@ fun <DFAState> DFA<DFAState>.withStates(retain: Set<DFAState>): DFA<DFAState> {
     return SimpleDFA(
         fullDFA.getStartingState(),
         productions,
-        retain.filter { fullDFA.isAccepting(it) }.toSet(),
+        retain.mapNotNull {
+            val result = fullDFA.result(it)
+            if (result != null) it to result else null
+        }.toMap(),
     )
 }
 
```

Comment:
I think we should also generalize infix `via` operator to also work with any `Atom`

Reply:
And maybe move this operator outside of minimalization directory, but this can be done during refactoring

Reply:
Changed to any Atom and added comment to refactor